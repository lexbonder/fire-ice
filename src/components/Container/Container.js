import React, { Component } from 'react';
import { connect } from 'react-redux';
import { Card } from '../Card/Card';
import wolf from '../../wolf.gif';
import PropTypes from 'prop-types';


export class Container extends Component {

  housesToRender = () => {
    const { houses } = this.props;
    if (!houses.length) {
      return <img src={wolf} alt='wolf running' />;
    } else {
      return houses.map( house => <Card key={house.url} houseInfo={house} />);
    }
  }

  render() {
    return (
      <div className='Container'>
        {this.housesToRender()}
      </div>
    );
  }
}

const { arrayOf, shape, string } = PropTypes;

Container.propTypes = {
  houses: arrayOf(
    shape({
      url: string,
      name: string,
      coatOfArms: string,
      words: string,
      titles: arrayOf(string),
      seats: arrayOf(string),
      founded: string,
      ancestralWeapons: arrayOf(string)
    })
  )
};

export const mapStateToProps = ({houses}) => ({houses});

export default connect(mapStateToProps)(Container);
