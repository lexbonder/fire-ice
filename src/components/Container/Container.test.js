import React from 'react';
import { shallow } from 'enzyme';
import { Container } from './Container';

describe('Container', () => {

  describe('housesToRender', () => {
    it('should match the snapshot if it gets an array of houses from props', () => {
      const mockHouses = [
        { house: 'yay', url: '1' },
        { house: 'fun times', url: '2' }
      ]
      const wrapper = shallow(<Container houses={mockHouses}/>)
      expect(wrapper).toMatchSnapshot()
    })

    it('should match the snapshot if it gets an empty array from props', () => {
      const mockHouses = []
      const wrapper = shallow(<Container houses={mockHouses}/>)
      expect(wrapper).toMatchSnapshot()
    })
  })
})