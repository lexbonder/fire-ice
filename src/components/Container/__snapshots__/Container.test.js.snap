// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Container housesToRender should match the snapshot if it gets an array of houses from props 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Container
    houses={
        Array [
            Object {
              "house": "yay",
              "url": "1",
            },
            Object {
              "house": "fun times",
              "url": "2",
            },
          ]
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Unknown
          houseInfo={
                    Object {
                              "house": "yay",
                              "url": "1",
                            }
          }
/>,
        <Unknown
          houseInfo={
                    Object {
                              "house": "fun times",
                              "url": "2",
                            }
          }
/>,
      ],
      "className": "Container",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": "1",
        "nodeType": "function",
        "props": Object {
          "houseInfo": Object {
            "house": "yay",
            "url": "1",
          },
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "2",
        "nodeType": "function",
        "props": Object {
          "houseInfo": Object {
            "house": "fun times",
            "url": "2",
          },
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Unknown
            houseInfo={
                        Object {
                                    "house": "yay",
                                    "url": "1",
                                  }
            }
/>,
          <Unknown
            houseInfo={
                        Object {
                                    "house": "fun times",
                                    "url": "2",
                                  }
            }
/>,
        ],
        "className": "Container",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": "1",
          "nodeType": "function",
          "props": Object {
            "houseInfo": Object {
              "house": "yay",
              "url": "1",
            },
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "2",
          "nodeType": "function",
          "props": Object {
            "houseInfo": Object {
              "house": "fun times",
              "url": "2",
            },
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`Container housesToRender should match the snapshot if it gets an empty array from props 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Container
    houses={Array []}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <img
        alt="wolf running"
        src="wolf.gif"
/>,
      "className": "Container",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "alt": "wolf running",
        "src": "wolf.gif",
      },
      "ref": null,
      "rendered": null,
      "type": "img",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <img
          alt="wolf running"
          src="wolf.gif"
/>,
        "className": "Container",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "alt": "wolf running",
          "src": "wolf.gif",
        },
        "ref": null,
        "rendered": null,
        "type": "img",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
